internal/vfs/utilities.go

```diff
@@ -248,7 +248,7 @@ func getIncludeBasePath(absolute string) string {
 			return tspath.RemoveTrailingDirectorySeparator(tspath.GetDirectoryPath(absolute))
 		}
 	}
-	return absolute[:strings.LastIndex(absolute, string(tspath.DirectorySeparator))]
+	return absolute[:strings.LastIndex(absolute[:wildcardOffset], string(tspath.DirectorySeparator))]
```

Comment:
The original code was:

```ts
function getIncludeBasePath(absolute: string): string {
    const wildcardOffset = indexOfAnyCharCode(absolute, wildcardCharCodes);
    if (wildcardOffset < 0) {
        // No "*" or "?" in the path
        return !hasExtension(absolute)
            ? absolute
            : removeTrailingDirectorySeparator(getDirectoryPath(absolute));
    }
    return absolute.substring(0, absolute.lastIndexOf(directorySeparator, wildcardOffset));
}
```

So, this looks correct to me; we need the last index starting at `wildcardOffset`. This slice cuts off the end, so is still a valid index for `absolute` itself.

That being said, you can't negative index a slice in Go, so the thing I'd wonder here is if the `LastIndex` is guaranteed to always succeed.

Reply:
Hmm yeah, using the debugger I just forced `absolute` to be a string with a `*` but no `/` and it does indeed panic.  But I don't think that could happen in any real scenario because `absolute` is from 
https://github.com/microsoft/typescript-go/blob/e8d352c89b733779275a364d6e5de7df65ac9a42/internal/vfs/utilities.go#L266-L270
and as a real absolute path (as far as I can tell), I think it _must_ start with `/`.

I'm very new to go, so I could be totally wrong and will happily close this or add more safety as you think is necessary.

Reply:
Yeah, it's possible that it's possible. The safe thing would be to write `min(strings.LastIndex(absolute[:wildcardOffset], string(tspath.DirectorySeparator)), 0)` but, I don't know that it matters.

Sort of surprised no tests change, but I guess we don't have a test for this.

Reply:
Done, I used `max` rather than `min` as `min` would still give us a negative slice, or always give us `0`

Reply:
Oh, yes, oops. 😄